### ISSUE TITLE
Marketplace UI/UX Improvements

### ISSUE THREAD

**Issue by thebkspecial** [Passage-Chain/community#5]
# Request for Proposal: Passage Marketplace Improvements
## 1. Overview
We are seeking proposals to redesign the existing Passage NFT marketplace and incorporate advanced features including NFT staking and comprehensive sales statistics. The objective is to improve and enhance user experience, increase engagement with Passage Chain, and provide robust tools for NFT traders.

### Marketplace UI/UX Improvements
Improve the Passage Marketplace UI/UX to be more intuitive, responsive, and visually appealing, with a focus on mobile support. 

#### Requirements:
- Full mobile compatibility/functionality 

#### Other desired features and improvements:
- Sorting by NFT collections, traits and/or price (both marketplace and user collection)
- Collection offers
- Offers on specific listed or unlisted NFTs
- Full functionality with Keplr and Leap Wallet on PC and mobile devices
- Auctions
- Sniping by traits
- Alerts when NFTs with certain traits and/or price are listed
- Ability for Passage team to define a certain sorting for default marketplace view
- Search by NFT number
- Bulk NFT trades

#### References: 
- [NFT Staking requirements](https://github.com/Passage-Chain/community/blob/main/sig-nft-staking/nft-staking-req.md) 
- [Existing Passage NFT vault contracts](https://github.com/envadiv/passage-contracts/tree/main/contracts/nft/nft-vault)

**Implement a staking mechanism that leverages the existing Passage NFT vault contracts, including:**
- Minimum staking period
- Unstaking duration
- Stake NFT
  - Moves NFT to vault contract
  - Can’t transfer or list to sell
- Unstake NFT
- Restake NFT during unstaking phase
- Hooks for calling other contracts upon completion of staking

**Create a new vault contract that allows the following:**
- Stake NFTs from multiple contracts
  - NFTs cannot be unstaked separately

**Create a contract to do the following:**
- Instantiate vaults based on the requirements of a creator (single collection staking or multiple collection)
- Emit IBC tokens deposited into the contract at setup at a specified rate
  - Tokens deposited by creator
  - Creator can add tokens after setup but not withdraw
  - Tokens are distributed at a set rate evenly across all staked NFTs

**Create an interface or frontend for setting up staking vaults. Access should be whitelisted by wallet.**
- Specify collection(s) to stake
- Set minimum staking period
- Set unstaking period
- Set tokens to be distributed
- Set rate of token distribution
- Deposit tokens

### Sales Data
Develop a sales statistics dashboard or analytics section that provides detailed insights into NFT sales, trends, and performance metrics, with real-time analytics capabilities. 

Desired features:
- Listings, sortable by collection, traits, and/or price
- Sales, sortable by collection, traits, and/or price
- Stats of sales over adjustable time scale (number of sales and revenue)
- Rarity of NFTs by collection and/or traits

## 3. Requirements
In addition to delivering on the scope of work outlined above, selected contributor(s) are expected to:

- Design UI/UX wireframes, mockups, and prototypes for approval (as appropriate, for frontend features)
- Test new features rigorously to ensure functionality and security
- Create documentation, including technical specifications, user guides, and maintenance procedures

## 4. Proposal Submissions
**Proposals can be made for the entire Scope of Work or individual projects including:**
- Marketplace redesign & mobile support
- NFT staking contracts
- NFT staking frontend/interface
- Statistics and analytics

**Interested teams and individuals should submit a detailed proposal including:**
- Background of submitting team/individual and relevant experience
- Approach and methodology for the project, including phases/milestones
- Cost estimate and breakdown of fees
- Examples/references from similar projects
- Timeline (project kickoff, milestones, and anticipated completion)

**Proposals will be evaluated based on:**
- Experience and expertise in NFT marketplace design and development
- Clarity and comprehensiveness of the proposed approach
- Cost-effectiveness and value
- Congruity of proposed work to the desired feature set

**Deadline for submitting proposals:**
Please submit your proposal by EOD Tuesday, July 23 for review and discussion during the [Passage Steering Committee meeting](https://github.com/Passage-Chain/community/blob/main/committee-steering/README.md) on July 24.

**For inquiries and submissions, please post on the relevant Github project:** 
- [Marketplace UI/UX Improvements](https://github.com/orgs/Passage-Chain/projects/1/views/1?pane=issue&itemId=71157404)
- [NFT Staking Smart Contracts](https://github.com/orgs/Passage-Chain/projects/1/views/1?pane=issue&itemId=61037914)
- [NFT Staking Frontend/Interface](https://github.com/orgs/Passage-Chain/projects/1/views/1?pane=issue&itemId=69427880)
- [Marketplace Sales Data](https://github.com/orgs/Passage-Chain/projects/1/views/1?pane=issue&itemId=71157471)

### 5. Budget
The following budget is an approximate range based on scope. Competitive bids will take into consideration quality and scope of work.

- **Marketplace UI/UX Improvements:** $10,000
- **NFT Staking Smart Contracts:** $5,000
- **NFT Staking Frontend/Interface:** $5,000
- **Marketplace Sales Data:** $5,000

**Total Estimated Budget: $20,000 - $30,000**

All bounties will be paid in PASG.

**Comment by Myestery** [comment]
@KodeSage We have worked on things like this before,
Can we get a figma design for the proposal before July 23rd?

**Comment by KodeSage** [comment]
yeah sure , we can get our designer to get low fidelity designs before 23rd of July but this is something we can pull off.

**Comment by KodeSage** [comment]
@Myestery

**Comment by KodeSage** [comment]
> @KodeSage We have worked on things like this before, Can we get a figma design for the proposal before July 23rd?

yeah sure , we can get our designer to get low fidelity designs before 23rd of July but this is something we can pull off.

**Comment by Myestery** [comment]
@thebkspecial Do we need to send the proposal to an email address or should we drop it here?

**Comment by BlockchainViper** [comment]
Is there a community discord? I follow on twitter but cannot find the discord.

**Comment by thebkspecial** [comment]
> Is there a community discord? I follow on twitter but cannot find the discord.

Discord is here! Check out the #council-chat channel https://discord.gg/passage

**Comment by thebkspecial** [comment]
> [@thebkspecial](https://github.com/thebkspecial) Do we need to send the proposal to an email address or should we drop it here?

Please drop proposals here! Happy to discuss details and give feedback on Discord before you post it here

**Comment by thebkspecial** [comment]
> > [@KodeSage](https://github.com/KodeSage) We have worked on things like this before, Can we get a figma design for the proposal before July 23rd?
> 
> yeah sure , we can get our designer to get low fidelity designs before 23rd of July but this is something we can pull off.

Designs are NOT needed for the proposal itself. Prior work samples should be sufficient. If you are selected, then we would ask for designs to be submitted

**Comment by BlockchainViper** [comment]
> yeah sure , we can get our designer to get low fidelity designs before 23rd of July but this is something we can pull off.

It seems like you don't need a UI/UX designer.

**Comment by thebkspecial** [comment]
> > yeah sure , we can get our designer to get low fidelity designs before 23rd of July but this is something we can pull off.
> 
> It seems like you don't need a UI/UX designer.

You would ideally have one if you are selected for the project.

**Comment by KodeSage** [comment]
> > > yeah sure , we can get our designer to get low fidelity designs before 23rd of July but this is something we can pull off.
> > 
> > 
> > It seems like you don't need a UI/UX designer.
> 
> You would ideally have one if you are selected for the project.



HI @thebkspecial  Good Day!, I left a Private message to on discord with our drafted proposal ready to be presented during tomorrow's call.

kindly please check your DM on discord. your feedback is very much appriciated. Thank you.

**Comment by aqt01** [comment]
#### Team

**Lowell Abbott - CEO** 
https://www.linkedin.com/in/lowell-abbott-vidal-8280aa72/

Former Devops Lead of Pocket Network and CEO of nodefleet.org. Has around 5 years of experience managing blockchain nodes and infrastructure systems in large scale , as well as leading devops/development teams. Crazy about optimising everything.


**Kael Abbott - CTO - Devops/Backend lead**
https://www.linkedin.com/in/kael-abbott-ab578713a/

Has a wide experience working on python/django backend systems and Devops (ansible, kubernetes, docker-compose) Previous experiences include working on https://elmomento.io/ with strategies on freqtrade, showing data statistics and working on high demanding backends on django/flask

**Steven Martinez - Devops/Backend engineer**
https://www.linkedin.com/in/albert-steven-mart%C3%ADnez-ferreira-933a09198

DevOp engineer with wide experience running blockchain nodes on different stacks such as ansible, docker-compose, kubernetes and experience on monitoring systems such as loki and grafana.


**Ordanis Sanchez - Full Stack dev**
https://www.linkedin.com/in/ordanis/

A great professional with previous experience on crypto and also startup companies as Full stack developer on rust/golang/react 

**Katherine Andujar - UI/UX Designer**
https://www.linkedin.com/in/katherine-and%C3%BAjar-843631121/

She's a web3 UI/UX professional designer currently working on nodefleet.org and other fintech startup products, author of the proposed explorer design.



#### Experience


Nodefleet infrastructure Around 45 different blockchains multi regionally and processing around 40-100m requests daily on bare metal machines fully optimized with failover in every region securing enterprise SLA. Founded by an ex infrastructure lead at pokt.network and 2 devops engineers with more than 4 years of experience in operating blockchain nodes.

We provide node and infrastructure services to several deping networks such as pokt.network, lava network, stateless API, DRPC and other protocols that are on testnet phase. We have also private RPC clients and are actively collaborating with providing vital maintenance on infrastructure parts and development of explorers, faucets, seeds nodes, archival nodes, validators etc.. More information below


### Grants

**Pokt.network Testnet and mainnet infrastructure maintenance on pokt.network**
**Maintenance of testnet/mainnet vital infrastructure**
https://forum.pokt.network/t/infrastructure-support-testnet-experiment/4092

POKTradar explorer
https://poktradar.io/home https://forum.pokt.network/t/pause-socket-for-poktradar-explorer/4870/7

Infrastructure maintenance node seed grant
https://forum.pokt.network/t/maintainer-grant-mainnet-infrastructure-seed-socket/4896

Passage archival node maintenance
https://x.com/passageio/status/1790449760333070542

### Collaborations and partnerships

Stateless API
https://x.com/statelessapi/status/1760786166536622227
We’ve been providing RPC to Stateless service since its beginnings.

DRPC
We are actively providing relays for the DRPC protocol across 25 chains since it's early start


**Staking**
We have an amount of around 6-8m USD in staking across $POKT, $ETH, $XAI, $LAVA and among others protocol blockchains in staked value and serving/participating critically on various DEPING networks

Pokt.network
We have around 45M $POKT staked in pokt,network (around 2m usd) with 14% of the validators and serving around 20% of the network
https://www.poktscan.com/node_runner?domain=nodefleet.org

Lava network worker
https://info.lavanet.xyz/provider/lava@1z28m49gczdtmqfdxkfdhnxpcmfmk3qmtlv5y0x#health

### Mobile app experience

We have designed/developed an app with collaboratoin with an ORG on the Dominican Republic for health purposes
https://play.google.com/store/apps/details?id=com.pillarware.hope&hl=es_MX


## Proposal and milestones


#### Statistics dashboard

- Phase 1
**UI/UX creation**
Timeline: 1-3 weeks
Our team will be designing the initial wireframe and UI for the web platform and mobile app with the desired features and improvements described in the proposal based on the top 3 NFT marketplaces in the market and inspirational input from the team to guide the desired UI design


- Phase 2
**UI/UX research and validation**
Timeline : 1-3 weeks

We'll test our UI/UX design with 10 beta testers, 2 from the core team of PASG, 2 from the community and the rest from other blockchain networks and communities in order to have better feedback and understand the user needs/interaction with our design. We also plan to open this as public tests later to gain community feeedback large

Maze it's a tool that we are going to use that we have used it before for validating the desing of poktradar.io

https://t.maze.co/217247809


- Phase 3
**Final test results and final adjustments UI/uX**
Timeline: 1-2 week

This final step will include a final report on the test taken and final adjustment on the design and final validation and delivery of the design based on the user research 


### Budget

#### Marketplace UI/UX Improvements
We can work on this project with a budget of 8k usd instead of 10k we have previous experience designing our own blockchain platform as well as other dashboard/demanding products since the UI/UX implenentation to the deployment of it


Link to proposal here:
https://docs.google.com/document/d/1-iJ3agwG8SF8_d9dn5bHVlck9N2Wx43q_K-Kxr1CzXM/edit?usp=drivesdk

**Comment by thebkspecial** [comment]
Confirming that the Steering Committee has awarded the following bounty for Marketplace UI/UX Improvements!

$8,000 to @aqt01 | [Link to Proposal](https://docs.google.com/document/d/1DD3_zNOYBmH4Jyqs4FacSjJOTKwA9XLuyDxK4I8yEVU/edit?tab=t.0#heading=h.2rokv3badd0b)

Bounties will be paid as follows: 1/2 up front + 1/2 upon completion. All payments will be made in PASG based on USD value of PASG on the date of transaction.
